prototype/migrations/script.py.mako

```diff
@@ -9,14 +12,35 @@ Create Date: ${create_date}
 # revision identifiers, used by Alembic.
 revision = ${repr(up_revision)}
 down_revision = ${repr(down_revision)}
+branch_labels = ${repr(branch_labels)}
+depends_on = ${repr(depends_on)}
 
 from alembic import op
 import sqlalchemy as sa
 ${imports if imports else ""}
 
-def upgrade():
-    ${upgrades if upgrades else "pass"}
+def upgrade(engine_name):
+    globals()["upgrade_%s" % engine_name]()
+
+
+def downgrade(engine_name):
+    globals()["downgrade_%s" % engine_name]()
+
+<%
+    from flask import current_app
+    db_names = [''] + list(current_app.config.get("SQLALCHEMY_BINDS").keys())
+%>
+
+## generate an "upgrade_<xyz>() / downgrade_<xyz>()" function
+## for each database name in the ini file.
+
+% for db_name in db_names:
```

Comment:
Isn't this going to be a problem, having migration file contain separate migrations for each database? I.e. if `SQLALCHEMY_BINDS` ever changes, or even differs between production setups or even local setups, then this approach won't work, right?